components/global/global.js

```diff
@@ -5,23 +5,30 @@ import {
   unmountComponentAtNode,
 } from 'react-dom';
 
+function setGlobalNoscroll(flag) {
+  const body = global.window.document.body;
+  flag
+    ? body.classList.add('ui-global_noscroll')
+    : body.classList.remove('ui-global_noscroll');
+}
+
 /**
  * Global component
  * React component for transportation of modals, lightboxes, loading bars... to document.body
  */
 class Global extends Component {
   componentDidMount() {
-    if (this.props.noscroll) {
-      global.window.document.body.classList.add('ui-global_noscroll');
-    }
-
     this.target = global.window.document.createElement('div');
     this.target.classList.add('ui-global');
     global.window.document.body.appendChild(this.target);
-    this.componentDidUpdate();
+    this.componentDidUpdate({ noscroll: false });
```

Comment:
I know that it was already like this, but can you extract `componentDidUpdate`'s logic into another function and then both `componentDidMount` and `componentDidUpdate` make use of the same code?

Otherwise looks like it's using React's lifecycle methods for custom purpose (which currently it is...)

Reply:
@mAiNiNfEcTiOn sure, for me this is also looks a bit weird. I'll change that ;)